cloudapp/src/app/user.service.ts

```diff
@@ -16,16 +16,21 @@ export class UserService {
     // get the requests from the link string in the entity object (if there is user info in it)
     // then get the user info from the user_primary_id or user_id or primary_id field and extract the address from the response
     users$ = (entities: Entity[]) => {
+        let calls = entities.filter(e=>['LOAN','USER'].includes(e.type))
+        .map(e=>e.type=='LOAN' ? this.userFromLoan(e.link) : this.getRequestFromAlma(e.link))
```

Comment:
Here I recommend filtering on the known entity types that you want to handle and retrieving the users accordingly.

Reply:
I would need to check the entity types and add the ones we need afterwards. I am sure about the 'Request' type, but still need to check the others.

Reply:
Makes sense- I added Requests ....

Reply:
I am testing your new changes with 'request'. The app seems to break when there is one or more requests which doesn't have any user attached to them and gives this error message: 'GEThttp://localhost:4200/almaws/v1/users/undefined'

Reply:
You're right- thanks! Committed a fix which handles requests without users.